packages/iota/test/validation/basicOutput.spec.ts

```diff
@@ -47,7 +51,27 @@ describe("Basic output validation", () => {
         ));
     });
 
-    it("should fail when one of the unlocks is of unsupported type", () => {
+    it("should fail when one of the unlock condition is of unsupported type", () => {
+        const basicOutput = cloneBasicOutput(mockBasicOutput);
+        basicOutput.unlockConditions[3] = {
+            type: STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE,
+            address: {
+                type: ED25519_ADDRESS_TYPE,
+                pubKeyHash: "0x6920b176f613ec7be59e68fc68f597eb3393af80f74c7c3db78198147d5f1f92"
+            }
+        };
+
+        const result = validateBasicOutput(basicOutput, protocolInfoMock);
+
+        expect(result.isValid).toEqual(false);
+        expect(result.errors).toBeDefined();
+        expect(result.errors?.length).toEqual(1);
+        expect(result.errors).toEqual(expect.arrayContaining(
+            ["Basic output unlock condition type of an unlock condition must define one of the following types: Address Unlock Condition, Storage Deposit Return Unlock Condition, Timelock Unlock Condition, Expiration Unlock Condition."]
+        ));
+    });
+
+    it("should fail when the unlock conditions count is larger than allowed", () => {
```

Comment:
We could use a generic phrase "should fail when the *** count is out of bounds" for all tests with this type of check. So we cover that we mean larger and smaller at the same time.